src/analysis/processing/qgsalgorithmfiledownloader.cpp

```diff
@@ -63,6 +66,26 @@ QgsFileDownloaderAlgorithm *QgsFileDownloaderAlgorithm::createInstance() const
 void QgsFileDownloaderAlgorithm::initAlgorithm( const QVariantMap & )
 {
   addParameter( new QgsProcessingParameterString( QStringLiteral( "URL" ), tr( "URL" ), QVariant(), false, false ) );
+
+  std::unique_ptr< QgsProcessingParameterEnum > methodParam = std::make_unique < QgsProcessingParameterEnum > (
+        QStringLiteral( "METHOD" ),
+        QObject::tr( "Method" ),
+        QStringList()
+        << QObject::tr( "GET" )
+        << QObject::tr( "POST" ),
+        false,
+        QStringLiteral( "GET" )
+      );
+  methodParam->setHelp( QObject::tr( "The HTTP method to use for the request" ) );
```

Comment:
@Gustry is this a new feature in Processing or just an old feature nobody uses? From a user (and writer) pov, this could be quite helpful in many contexts

Reply:
It's indeed a new "mini" feature. Before, all requests were "GET". Now, you can use POST and also add a BODY parameter when it's POST.

For instance, you can make a POST query to the OSM Overpass server, with a very long query in the BODY. Before, with GET, you might have reached the limit of characters in a GET request.

Is-that OK ?

Reply:
Hi @Gustry Actually I was talking about the ability to add a description to a parameter (the setHelp method). I'm not sure I noticed it earlier in another algorithm

Reply:
Ah sorry.

It's a new feature from QGIS 3.16:
https://qgis.org/en/site/forusers/visualchangelog316/index.html#add-help-string-for-parameters

Not very used, but quite convenient indeed to add a longer description instead of adding a lot of text in the main algorithm description.